### PR TITLE
Time spine move from dbt-utils to dbt-core

### LINKED ISSUE #4124: Update metricflow_time_spine creation documentation to use dbt-core macro (from dbt-core PR #8616)
In https://github.com/dbt-labs/dbt-core/pull/8616 we moved the `date_spine` macro from `dbt-utils` to `dbt-core`. Thus the [documentation for creating the metricflow time spine](https://docs.getdbt.com/docs/build/metricflow-time-spine) should be updated to use the `dbt-core` built in macro instead of the `dbt-utils` macro

### PR DESCRIPTION
**Pull request by matthewshaver**
Previews:
- [1.6](https://docs-getdbt-com-git-time-spine-dbt-labs.vercel.app/docs/build/metricflow-time-spine?version=1.6)
- [1.7](https://docs-getdbt-com-git-time-spine-dbt-labs.vercel.app/docs/build/metricflow-time-spine?version=1.7)

## What are you changing in this pull request and why?

Closes: https://github.com/dbt-labs/docs.getdbt.com/issues/4124

Move time_spine from dbt-utils to dbt-core

Uncomment if you're publishing docs for a prerelease version of dbt (delete if not applicable):
- [x] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [x] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/guides/migration/versions)

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [x] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

### CODE REVIEW

**Comment by matthewshaver** [comment]
@dbeatty10 can you tell me if this is an accurate way to represent this change?

**Comment by dbeatty10** [comment]
> @dbeatty10 can you tell me if this is an accurate way to represent this change?

Use `dbt` instead of `dbt_core`, and then you'll be good-to-go!

File: website/docs/docs/build/metricflow-time-spine.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ To create this table, you need to create a model in your dbt project called `met
 with days as (
 
     {{
-        dbt_utils.date_spine(
+        dbt_core.date_spine(
```
Comment (matthewshaver) [inline website/docs/docs/build/metricflow-time-spine.md]:
```suggestion
        dbt.date_spine(
```

File: website/docs/docs/build/metricflow-time-spine.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ select * from final
 -- BigQuery supports DATE() instead of TO_DATE(). Use this model if you're using BigQuery
 {{config(materialized='table')}}
 with days as (
-    {{dbt_utils.date_spine(
+    {{dbt_core.date_spine(
```
Comment (matthewshaver) [inline website/docs/docs/build/metricflow-time-spine.md]:
```suggestion
    {{dbt.date_spine(
```

**Review by dbeatty10 — COMMENTED** [review]
I think this will only work for dbt v1.7+

@Jstein77 or @QMalcolm do we need to preserve instructions the use `dbt_utils.date_spine` for dbt v1.6?

**Comment by QMalcolm** [comment]
> I think this will only work for dbt v1.7+
> 
> 
> 
> @Jstein77 or @QMalcolm do we need to preserve instructions the use `dbt_utils.date_spine` for dbt v1.6?

That is correct

**Comment by matthewshaver** [comment]
> > I think this will only work for dbt v1.7+
> > @Jstein77 or @QMalcolm do we need to preserve instructions the use `dbt_utils.date_spine` for dbt v1.6?
> 
> That is correct

Corrected this based on the feedback here. Added version blocks

File: website/docs/docs/build/metricflow-time-spine.md [hunk 3]
```diff
@@ -38,8 +40,45 @@ final as (
...
+{{
+    config(
+        materialized = 'table',
+    )
+}}
+
+with days as (
+
+    {{
+        dbt.date_spine(
```
Comment (dbeatty10) [inline website/docs/docs/build/metricflow-time-spine.md:58]: Remove duplicated line:
```suggestion
```

**Review by runleonarun — APPROVED** [review]
:shipit:

**Pull request merged**